### PR TITLE
Fix empty table issue with s3

### PR DESCRIPTION
**Pull request by ravjotbrar**
### Summary

When empty table is exported to S3, no file is created. Fixed this issue by checking if file exists before attempting to read.

### Description

Used an if then else statement to check if file exists before calling getFileList. Also updated filestorelayer mock and tests that use the mock to reflect the change.

### Related Issue

http://jira.verticacorp.com:8080/jira/browse/VER-77447

### Additional Reviewers

@jonathanl-bq 
@alexr-bq

### CODE REVIEW

**Comment by alexr-bq** [comment]
Resolve conflicts

File: connector/src/test/scala/com/vertica/spark/datasource/core/VerticaDistributedFilesystemReadPipeTests.scala [hunk 1]
```diff
@@ -295,7 +297,7 @@ class VerticaDistributedFilesystemReadPipeTests extends AnyFlatSpec with BeforeA
       case Left(_) => fail
       case Right(partitionInfo) =>
         val partitions = partitionInfo.partitionSeq
-        assert(partitions.length == 3)
+        assert(partitions.length == 3||partitions.length == 0)
```
Comment (jonathanl-bq) [inline connector/src/test/scala/com/vertica/spark/datasource/core/VerticaDistributedFilesystemReadPipeTests.scala:300]: nit: add spaces around the || operator

File: connector/src/test/scala/com/vertica/spark/datasource/core/VerticaDistributedFilesystemReadPipeTests.scala [hunk 2]
```diff
@@ -401,7 +405,7 @@ class VerticaDistributedFilesystemReadPipeTests extends AnyFlatSpec with BeforeA
       case Left(_) => fail
       case Right(partitionInfo) =>
         val partitions = partitionInfo.partitionSeq
-        assert(partitions.length == partitionCount)
+        assert(partitions.length == partitionCount || partitions.length==0)
```
Comment (jonathanl-bq) [inline connector/src/test/scala/com/vertica/spark/datasource/core/VerticaDistributedFilesystemReadPipeTests.scala:408]: nit: spaces around ==

File: connector/src/test/scala/com/vertica/spark/datasource/core/VerticaDistributedFilesystemReadPipeTests.scala [hunk 3]
```diff
@@ -448,7 +453,7 @@ class VerticaDistributedFilesystemReadPipeTests extends AnyFlatSpec with BeforeA
...
```
Comment (jonathanl-bq) [inline connector/src/test/scala/com/vertica/spark/datasource/core/VerticaDistributedFilesystemReadPipeTests.scala:456]: same here

**Review by jonathanl-bq — APPROVED** [review]

File: functional-tests/build.sbt [hunk 4]
```diff
@@ -34,7 +34,7 @@ libraryDependencies += "org.scalatest" %% "scalatest" % "3.2.2" % "test"
 libraryDependencies += "com.typesafe.scala-logging" %% "scala-logging" % "3.9.2"
 libraryDependencies += "org.scalamock" %% "scalamock" % "4.4.0" % Test
 libraryDependencies += "org.typelevel" %% "cats-core" % "2.3.0"
-libraryDependencies += "org.apache.hadoop" % "hadoop-aws" % "3.2.0"
+libraryDependencies += "org.apache.hadoop" % "hadoop-aws" % "3.3.0"
```
Comment (alexr-bq) [inline functional-tests/build.sbt:37]: this might break the integration tests. if not then all good

**Review by alexr-bq — APPROVED** [review]

**Review by alexr-bq — DISMISSED** [review]

**Pull request merged**